src/main/core/worker.rs

```diff
@@ -270,6 +274,106 @@ impl Worker {
         .unwrap();
     }
 
+    pub fn send_packet(src_host: &Host, packet: *mut cshadow::Packet) {
```

Comment:
Nit - I'd consider this `unsafe`. Even though the Rust code doesn't itself dereference the `packet`, it's still the caller's responsibility that `packet` is valid and not being accessed  simultaneously from other threads to avoid UB in the C code

Reply:
Added.